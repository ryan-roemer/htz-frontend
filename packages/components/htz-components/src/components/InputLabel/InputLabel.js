import React from 'react';
import PropTypes from 'prop-types';
import { createComponent, } from 'react-fela';

const propTypes = {
  /** id of input passed to html label "for" attribute */
  labelFor: PropTypes.string,
  /** text caption for the relevant input */
  text: PropTypes.string,
  /** child node, should be an input that the label is connected to */
  children: PropTypes.node,
  /** general style of component, insert a regular css in js object, not limited to color and font-size */
  styleObject: PropTypes.shape({
    color: PropTypes.string,
    fontSize: PropTypes.string,
  }),
};
const defaultProps = {
  text: undefined,
  labelFor: undefined,
  children: undefined,
  styleObject: undefined,
};

const labelStyle = ({ styleObject, }) => ({
  ...styleObject,
});

const StyledLabel = createComponent(labelStyle, 'label', [ 'htmlFor', ]);

function InputLabel(props) {
  return (
    <StyledLabel
      {...props.labelFor && { htmlFor: props.labelFor, }}
      styleObject={props.styleObject}
    >
      {props.text && <span>{props.text}</span>}
      {props.children}
    </StyledLabel>
  );
}

InputLabel.propTypes = propTypes;
InputLabel.defaultProps = defaultProps;

export default InputLabel;
