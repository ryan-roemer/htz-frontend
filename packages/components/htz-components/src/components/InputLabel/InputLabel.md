<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**

- [A basic input label component](#a-basic-input-label-component)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

### A basic input label component

Can be used to create an input label, that is connected to an input with htmlFor, or by having a child input component

A basic input label
```jsx 
<InputLabel 
 text="This a label"
 />
```

Label with a style properties ,labelFor, and text props
```jsx 
     <InputLabel
        labelFor="aa"
        text="this is a test text passed through a prop"
        styleObject= {
          {
            fontSize: '4rem',
            color: 'blue',
            backgroundColor: 'red',
          }
        }
      />;
```






