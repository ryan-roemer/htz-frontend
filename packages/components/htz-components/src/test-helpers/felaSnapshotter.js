import React from 'react';
import createRenderer from '../components/StyleProvider/createRenderer';
import StyleProvider from '../components/StyleProvider/StyleProvider';
import createFelaSnapshotter from './createFelaSnapshotter';

const testRenderer = createRenderer({ isRtl: true, });
const felaSnapshotter = createFelaSnapshotter(StyleProvider, testRenderer);

export default felaSnapshotter;
