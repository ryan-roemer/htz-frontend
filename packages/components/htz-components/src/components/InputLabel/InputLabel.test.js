import React from 'react';
import felaSnapshotter from '../../test-helpers/felaSnapshotter';
import InputLabel from './InputLabel';

it('should correctly render a InputLabel without any props or styles', () => {
  const snapshot = felaSnapshotter(<InputLabel />);
  expect(snapshot).toMatchSnapshot();
});
it('should correctly render a InputLabel with custom text passed as a prop ', () => {
  const snapshot = felaSnapshotter(<InputLabel text="this is a test text passed through a prop" />);
  expect(snapshot).toMatchSnapshot();
});
it('should correctly render a InputLabel with custom labelFor passed as a prop ', () => {
  const snapshot = felaSnapshotter(<InputLabel labelFor="ID12345" />);
  expect(snapshot).toMatchSnapshot();
});

it('should render a Input Label with styles and text matching styleObject and text props', () => {
  const snapshot = felaSnapshotter(
    <InputLabel
      text="a text prop"
      styleObject={{
        fontSize: '4rem',
        color: 'blue',
        backgroundColor: 'red',
      }}
    />
  );
  expect(snapshot).toMatchSnapshot();
});
