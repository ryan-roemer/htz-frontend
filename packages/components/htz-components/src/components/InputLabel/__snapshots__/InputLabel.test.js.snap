// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render a InputLabel with custom labelFor passed as a prop  1`] = `
Object {
  "component": <div>
    <label
      className=""
      htmlFor="ID12345"
    />
  </div>,
  "styles": "",
}
`;

exports[`should correctly render a InputLabel with custom text passed as a prop  1`] = `
Object {
  "component": <div>
    <label
      className=""
    >
      <span>
        this is a test text passed through a prop
      </span>
    </label>
  </div>,
  "styles": "",
}
`;

exports[`should correctly render a InputLabel without any props or styles 1`] = `
Object {
  "component": <div>
    <label
      className=""
    />
  </div>,
  "styles": "",
}
`;

exports[`should render a Input Label with styles and text matching styleObject and text props 1`] = `
Object {
  "component": <div>
    <label
      className="htz-a htz-b htz-c"
    >
      <span>
        a text prop
      </span>
    </label>
  </div>,
  "styles": ".htz-a{font-size:4rem}.htz-b{color:blue}.htz-c{background-color:red}",
}
`;
